src/components/molecules/BalanceGameBox/BalanceGameBox.tsx

```diff
@@ -1,47 +1,100 @@
 import React, { useState } from 'react';
-import {
-  BalanceGameButton,
-  BalanceGameButtonProps,
-} from '@/components/atoms/BalanceGameButton/BalanceGameButton';
+import { GameOption } from '@/types/game';
+import { getRandomNumbers } from '@/utils/calculator';
+import BalanceGameButton from '@/components/atoms/BalanceGameButton/BalanceGameButton';
+import { useCreateGameVoteMutation } from '@/hooks/api/vote/useCreateGameVoteMutation';
+import { useEditGameVoteMutation } from '@/hooks/api/vote/useEditGameVoteMutation';
+import { useDeleteGameVoteMutation } from '@/hooks/api/vote/useDeleteGameVoteMutation';
 import * as S from './BalanceGameBox.style';
 
 export interface BalanceGameBoxProps {
-  values: BalanceGameButtonProps[];
+  gameSetId: number;
+  gameId: number;
+  options?: GameOption[];
+  selectedVote: 'A' | 'B' | null;
+  handleNextStage: () => void;
 }
 
-export const BalanceGameBox = ({ values }: BalanceGameBoxProps) => {
-  const [selectedButton, setSelectedButton] = useState<'A' | 'B' | null>(null);
+const BalanceGameBox = ({
+  gameSetId,
+  gameId,
+  options,
+  selectedVote,
+  handleNextStage,
+}: BalanceGameBoxProps) => {
+  const optionA = options?.[0];
+  const optionB = options?.[1];
 
-  const handleButtonClick = (optionType: 'A' | 'B') => {
-    if (selectedButton === optionType) {
-      setSelectedButton(null);
+  const getRandomImages = () => {
+    const [randomNumberA, randomNumberB] = getRandomNumbers(
+      S.gameBgArray.length,
+    );
+    return [S.gameBgArray[randomNumberA], S.gameBgArray[randomNumberB]];
+  };
+
+  const [backgroundImages] = useState<string[]>(getRandomImages);
+  const [backgroundImageA, backgroundImageB] = backgroundImages;
+
+  const { mutate: createGameVote } = useCreateGameVoteMutation(
+    gameSetId,
+    gameId,
+  );
+  const { mutate: editGameVote } = useEditGameVoteMutation(gameSetId, gameId);
+  const { mutate: deleteGameVote } = useDeleteGameVoteMutation(
+    gameSetId,
+    gameId,
+  );
+
+  const handleGameVote = (
+    selectedOption: 'A' | 'B' | null,
+    voteOption: 'A' | 'B',
+  ) => {
+    if (!selectedOption) {
+      createGameVote(voteOption, {
+        onSuccess: () => {
+          setTimeout(() => {
+            handleNextStage();
+          }, 700);
+        },
+      });
+    } else if (selectedOption === voteOption) {
+      deleteGameVote();
     } else {
-      setSelectedButton(optionType);
+      editGameVote(voteOption);
     }
   };
 
-  const optionA = values[0];
-  const optionB = values[1];
+  const handleButtonClick = (voteOption: 'A' | 'B') => {
+    handleGameVote(selectedVote, voteOption);
+  };
 
   return (
-    <div css={S.containerStyle}>
+    <div css={[S.containerStyle, S.getOutlineStyle(selectedVote ?? null)]}>
```

Comment:
```suggestion
    <div css={[S.containerStyle, S.getOutlineStyle(selectedVote)]}>
```
로 처리해줘도 동일한 효과를 적용할 수 있지 않을까 라는 의견을 제시해봅니다!

가능하면 한번 테스트도 부탁드려요 :)